lib/persistence/dao/armour_dao.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:battle_it_out/persistence/dao/dao.dart';
+import 'package:battle_it_out/persistence/dao/item_quality_dao.dart';
+import 'package:battle_it_out/persistence/entities/armour.dart';
+import 'package:battle_it_out/persistence/entities/item_quality.dart';
+import 'package:battle_it_out/persistence/database_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+class ArmourDAO extends DAO<Armour> {
+  @override
+  get tableName => 'armour';
+
+  Future<List<ItemQuality>> getQualities(int id) async {
+    Database? database = await DatabaseProvider.instance.getDatabase();
+
+    final List<Map<String, dynamic>> map =
+        await database!.query("armour_qualities", where: "ARMOUR_ID = ?", whereArgs: [id]);
+    List<ItemQuality> qualities = [];
```

Comment:
you could probably do something like:
var qualities = [for (var item in map.values) await ItemQualityDAO().get(item["QUALITY_ID"])]